### PR TITLE
Add/move examples

### PR DESCRIPTION
**Pull request by impronunciable**
- Moved examples to the folder
- Added readme with install + explanation
- Added examples from wiki

### CODE REVIEW

File: examples/with-styled-components/pages/_document.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export default class MyDocument extends Document {
+  static async getInitialProps (ctx) {
+    const renderPage = () => {
+      return ctx.renderPage()
+    }
+
+    const props = await Document.getInitialProps({ ...ctx, renderPage })
+    const style = styleSheet.rules().map(rule => rule.cssText).join('\n')
+    return { ...props, style }
+  }
```
Comment (nkzawa) [inline examples/with-styled-components/pages/_document.js]: `Document.getInitialProps` includes the implementation of `styled-jsx` and `next/css`, so instead I think it should be like the following:

```js
static async getInitialProps ({ renderPage }) {
    const page = renderPage()
    const style = styleSheet.rules().map(rule => rule.cssText).join('\n')
    return { ...page, style }
}
```

Comment (nkzawa) [inline examples/with-styled-components/pages/_document.js]: @rauchg I noticed all examples on Wiki are calling `Document.getInitialProps`. What do you think about this API ?

Comment (rauchg) [inline examples/with-styled-components/pages/_document.js]: @nkzawa sweet :)

Comment (rauchg) [inline examples/with-styled-components/pages/_document.js]: although we need to call .flush() before `renderPage` to reset

**Pull request merged**